presto-accumulo/src/main/java/com/facebook/presto/accumulo/model/AccumuloSplit.java

```diff
@@ -0,0 +1,270 @@
+import com.fasterxml.jackson.databind.JsonDeserializer;
+import com.fasterxml.jackson.databind.JsonSerializer;
+import com.fasterxml.jackson.databind.SerializerProvider;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import com.google.common.collect.ImmutableList;
+import org.apache.accumulo.core.data.Range;
+import org.apache.hadoop.io.DataInputBuffer;
+import org.apache.hadoop.io.DataOutputBuffer;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Optional;
+
+import static com.facebook.presto.accumulo.AccumuloErrorCode.VALIDATION;
+import static com.google.common.base.MoreObjects.toStringHelper;
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+/**
+ * Jackson object and an implementation of a Presto ConnectorSplit. Encapsulates all data regarding
+ * a split of Accumulo table for Presto to scan data from.
+ */
+public class AccumuloSplit
+        implements ConnectorSplit
+{
+    private final String connectorId;
+    private final Optional<String> hostPort;
+    private final String rowId;
+    private final String schema;
+    private final String table;
+    private final String serializerClassName;
+    private final Optional<String> scanAuthorizations;
+    private final List<HostAddress> addresses;
+
+    @JsonSerialize(contentUsing = RangeSerializer.class)
+    @JsonDeserialize(contentUsing = RangeDeserializer.class)
+    private List<Range> ranges;
+    private final List<AccumuloColumnConstraint> constraints;
+
+    @JsonCreator
+    public AccumuloSplit(@JsonProperty("connectorId") String connectorId,
+            @JsonProperty("schema") String schema,
+            @JsonProperty("table") String table,
+            @JsonProperty("rowId") String rowId,
+            @JsonProperty("serializerClassName") String serializerClassName,
+            @JsonProperty("ranges") List<Range> ranges,
+            @JsonProperty("constraints") List<AccumuloColumnConstraint> constraints,
+            @JsonProperty("scanAuthorizations") Optional<String> scanAuthorizations,
+            @JsonProperty("hostPort") Optional<String> hostPort)
+    {
+        this.connectorId = requireNonNull(connectorId, "connectorId is null");
+        this.rowId = requireNonNull(rowId, "rowId is null");
+        this.schema = requireNonNull(schema, "schema is null");
+        this.table = requireNonNull(table, "table is null");
+        this.serializerClassName = requireNonNull(serializerClassName, "serializerClassName is null");
+        this.constraints = requireNonNull(constraints, "constraints is null");
+        this.scanAuthorizations = requireNonNull(scanAuthorizations, "scanAuthorizations is null");
+        this.hostPort = requireNonNull(hostPort, "hostPort is null");
+
+        // We don't "requireNotNull" this field, Jackson parses objects using a top-down approach,
```

Comment:
I don't understand why this is necessary. Can you write a test for this class that verifies the JSON serialization? For example, see `TestHiveSplit`.